### ISSUE TITLE
Oculus Quest 2: Crashes (sigsegv) the app on null dereferences

### ISSUE THREAD

**Issue by johanhelsing-attensi** [getsentry/sentry-unity#499]
### Environment

How do you use Sentry?
sentry.io

Which SDK and version?
0.9.1

### Steps to Reproduce

Built an android app, installed it on oculus quest 2 and threw an exception.

### Expected Result

Error message is sent to sentry.io, app continues running

### Actual Result

No error sent. App quits instantaneously. Logcat shows this:

```
12-23 10:07:33.472 11702 11702 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
12-23 10:07:33.472 11702 11702 F DEBUG   : Build fingerprint: 'oculus/hollywood/hollywood:10/QQ3A.200805.001/21985310426300000:user/release-keys'
12-23 10:07:33.472 11702 11702 F DEBUG   : Revision: '0'
12-23 10:07:33.472 11702 11702 F DEBUG   : ABI: 'arm'
12-23 10:07:33.473 11702 11702 F DEBUG   : Timestamp: 2021-12-23 10:07:33+0100
12-23 10:07:33.473 11702 11702 F DEBUG   : pid: 11572, tid: 11604, name: UnityMain  >>> com.MyCompany.MyProject <<<
12-23 10:07:33.473 11702 11702 F DEBUG   : uid: 10095
12-23 10:07:33.473 11702 11702 F DEBUG   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0
12-23 10:07:33.473 11702 11702 F DEBUG   : Cause: null pointer dereference
12-23 10:07:33.473 11702 11702 F DEBUG   :     r0  00000000  r1  bb5583c0  r2  00000000  r3  00000000
12-23 10:07:33.473 11702 11702 F DEBUG   :     r4  00000000  r5  00000000  r6  00000000  r7  aaeec9c0
12-23 10:07:33.473 11702 11702 F DEBUG   :     r8  00000000  r9  cdafc360  r10 00000000  r11 cdafbce8
12-23 10:07:33.473 11702 11702 F DEBUG   :     ip  ebae6928  sp  cdafbcd0  lr  00000000  pc  a011dac0
12-23 10:07:33.921 11702 11702 F DEBUG   :
12-23 10:07:33.921 11702 11702 F DEBUG   : backtrace:
12-23 10:07:33.921 11702 11702 F DEBUG   :       #00 pc 0000dac0  <anonymous:a0110000>
```

Happened with Sentry 0.6.0 and 0.9.1.

If I remove sentry from my unity package, the app does not crash and continues as normal.

For what it's worth, I maintain a [fork of sentry-unity-lite](https://github.com/johanhelsing/sentry-unity-simple) with some quality of life improvements, and that also works without issues. So if anyone else runs into this, that's a viable workaround if you mostly care about script errors.

**Comment by bitsandfoxes** [comment]
Hey @johanhelsing-attensi. Thanks for raising this. I assume you let the SDK initialize itself and configure it via the editor window? Could you try the following settings
-  Transport 
  - Enable Offline Caching: Unchecked
- Advanced 
  - Enable Debug Output: Checked
  - Only in Editor: Unchecked
  - Verbosity Level: Debug
and post the log output here?

**Comment by johanhelsing-attensi** [comment]
Thanks for following up!

I looked further into it and found out a few things.

First, it seems to be mostly be related to null dereferences. If I simply throw an error, it does not happen.

The bug got somewhat complicated because the actual error I first experienced happened within a dotween callback. When I moved it outside, it still crashed, but I now see the lines:

```
01-05 09:54:16.732  8093  8239 I Unity   : Sentry: (Debug) Envelope '' sent successfully. Payload:
01-05 09:54:16.732  8093  8239 I Unity   : {"sdk":{"name":"Sentry","version":"3.9.3"}}
01-05 09:54:16.732  8093  8239 I Unity   : {"type":"session","length":294}
01-05 09:54:16.732  8093  8239 I Unity   : {"sid":"a976c1108031486ab7ff25ee90f4e50a","did":"334389048B872A533002B34D73F8C29FD09EFC50","init":true,"started":"2022-01-05T09:54:13.040481+01:00","timestamp":"2022-01-05T08:54:13.183927+00:00","seq":0,"duration":0,"errors":0,"attrs":{"release":"Interpersonal@0.1","environment":"production"}}
01-05 09:54:16.732  8093  8239 I Unity   :
01-05 09:54:16.732  8093  8239 I Unity   : UnityEngine.StackTraceUtility:ExtractStackTrace () (at /Users/bokken/buildslave/unity/build/Runtime/Export/Scripting/StackTrace.cs:37)
01-05 09:54:16.732  8093  8239 I Unity   : UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
01-05 09:54:16.732  8093  8239 I Unity   : UnityEngine.Logger:Log (UnityEngine.LogType,object)
01-05 09:54:16.732  8093  8239 I Unity   : UnityEngine.Debug:Log (object)
01-05 09:54:16.732  8093  8239 I Unity   : Sentry.Unity.UnityLogger:Log (Sentry.SentryLevel,string,System.Exception,object[]) (at /Users/runner/work/sentry-unity/sentry-unity/src/Sentry.Unity/UnityLogger.cs:39)
01-05 09:54:16.732  8093  8239 I Unity   : Sentry.Extensibility.DiagnosticLoggerExtensions:LogIfEnabled<System.Nullable`1<Sentry.SentryId>, string> (Sentry.
01-05 09:54:17.037  8255  8255 F DEBUG   :
01-05 09:54:17.037  8255  8255 F DEBUG   : backtrace:
01-05 09:54:17.037  8255  8255 F DEBUG   :       #00 pc 0000af8c  <anonymous:b83f0000>
01-05 09:54:17.506  8093  8093 I Unity   : onPause
01-05 09:54:17.551  1041  1301 I ActivityManager: Process com.Attensi.Interpersonal (pid 8093) has died: fore TOP
```

With that out of the way, I created a more minimal test case:

```c#
using UnityEngine;

public class ErrorOnStart : MonoBehaviour
{
    private float time;

    private void Start()
    {
        ErrorOnStart nullReference = null;
        nullReference.time = 123;
    }
}
```

With sentry installed, this causes the application to crash and quit, without I simply see:

```
01-05 10:04:43.189  8542  8564 E Unity   : NullReferenceException: Object reference not set to an instance of an object
01-05 10:04:43.189  8542  8564 E Unity   :   at ErrorOnStart.Start () [0x00003] in C:\Users\johhel\dev\unity_att_interpersonal\Assets\ErrorOnStart.cs:13
01-05 10:04:43.189  8542  8564 E Unity   :
01-05 10:04:46.077  8542  8564 I Unity   : VersionHandler: Local versions read. (74)
01-05 10:04:46.077  8542  8564 I Unity   : UnityEngine.StackTraceUtility:ExtractStackTrace () (at /Users/bokken/buildslave/unity/build/Runtime/Export/Scripting/StackTrace.cs:37)
01-05 10:04:46.077  8542  8564 I Unity   : UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
01-05 10:04:46.077  8542  8564 I Unity   : UnityEngine.Logger:Log (UnityEngine.LogType,object)
01-05 10:04:46.077  8542  8564 I Unity   : UnityEngine.Debug:Log (object)
01-05 10:04:46.077  8542  8564 I Unity   : Core.Utilities.VersionHandler/<ReadLocalVersions>d__27:MoveNext () (at C:/Users/johhel/dev/unity_att_interpersonal/Library/PackageCache/com.attensi.core.essentials@22.3.0/Runtime/Scripts/Backend/VersionHandler.cs:221)
01-05 10:04:46.077  8542  8564 I Unity   : Core.Utilities.Execute/StatefulCoroutine/<InternalRoutine>d__18:MoveNext () (at C:/Users/johhel/dev/unity_att_interpersonal/Library/PackageCache/com.attensi.core.base@2.4.1/Runtime/Execution/Execute.cs:664)
01-05 10:04:46.077  8542  8564 I Unity   : UnityEngine.SetupCoroutine:InvokeMoveNext (System.Collections.IEnumerator,intptr) (at /Users/bokken/buildslave/unity/build/Runtime/Export/Scripting/Coroutines.cs:17)
```

in the logs, and the app continues running.

So, TL;DR: Having sentry installed turns otherwise harmless null dereferences into fatal crashes.

[Logs with sentry-unity-simple (no crash)](https://github.com/getsentry/sentry-unity/files/7813609/sentry-simple-null-dereference.log)
[Logs with sentry-unity (this repo, crashes)](https://github.com/getsentry/sentry-unity/files/7813610/sentry-null-dereference-crash.log)

**Comment by bruno-garcia** [comment]
Thanks for the details. I have one of those home, so I'll try to repro this. 
Could you please let us know the Unity version you're using? Any other detail that can help us reproduce this like IL2CPP linker settings, etc? Maybe screenshots of the Player build configuration would be helpful.

**Comment by oddgames-david** [comment]
Possibly related, I have a pocophone that crashes in a similar fashion with auto mode turned on. 

I can get a demo project to run fine with the same unity version so it must be a conflict with other plugins in my project or perhaps one of the gradle files, still investigating.

Cache mode is off.

![image](https://user-images.githubusercontent.com/1932174/149060155-fe5fe4f4-7626-40d5-ad1d-4c8fe3f2f0de.png)

**Comment by oddgames-david** [comment]
Full log from process ID from a crash
[](url
[log from process id.txt](https://github.com/getsentry/sentry-unity/files/7852432/log.from.process.id.txt)
)

**Comment by johanhelsing** [comment]
> Could you please let us know the Unity version you're using? Any other detail that can help us reproduce this like IL2CPP linker settings, etc? Maybe screenshots of the Player build configuration would be helpful.

I'm on Unity 2020.3.22

I tried a *ton* of different settings (developer mode on/off, opengl/vulkan, graphics jobs) before I pinned it down on sentry being the culprit, but seemed to crash regardless of what settings I used. Pretty sure I used mono and armv7 the entire time, though.

I also think I made a build for my android phone, which seemed to crash in a similar way, though I'm not super sure about this since it's quite a while ago. In any case, it looks like it may be an android-wide bug, and not specific to oculus quest 2).

**Comment by johanhelsing-attensi** [comment]
> Could you please let us know the Unity version you're using? Any other detail that can help us reproduce this like IL2CPP linker settings, etc? Maybe screenshots of the Player build configuration would be helpful.

I'm on Unity 2020.3.22

I tried a *ton* of different settings (developer mode on/off, opengl/vulkan, graphics jobs) before I pinned it down on sentry being the culprit, but seemed to crash regardless of what settings I used. Pretty sure I used mono and armv7 the entire time, though.

I also think I made a build for my android phone, which seemed to crash in a similar way, though I'm not super sure about this since it's quite a while ago. In any case, it looks like it may be an android-wide bug, and not specific to oculus quest 2).

**Comment by bruno-garcia** [comment]
Seems related to: https://github.com/getsentry/sentry-java/issues/1867
We'll bump sentry-native and sentry-android and make a release

**Comment by bruno-garcia** [comment]
@oddgames-david You have this error using the same repro @johanhelsing-attensi described? The null deref?
And to confirm, this happens only on Mono on armv7? Not on IL2CPP? Not on Arm64?

@johanhelsing-attensi does this happen on IL2CPP and/or Mono? What device architecture are you trying?

**Comment by bitsandfoxes** [comment]
I'm able to reproduce the issue locally. We'll take a look.

**Comment by johanhelsing-attensi** [comment]
> @johanhelsing-attensi does this happen on IL2CPP and/or Mono? What device architecture are you trying?

I've only used sentry-unity on mono, so I don't know if il2cpp is different. Both devices were arm64.

**Comment by bruno-garcia** [comment]
We're releasing this fix in 0.9.4 coming out now
